app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/reliefdetails/ReliefDetailsController.scala

```diff
@@ -333,10 +333,10 @@ class ReliefDetailsController @Inject() (
           page = otherReliefsPage(_, _)
         )(
           requiredPreviousAnswer = _.fold(
-            _.lettingsRelief,
-            c => Some(c.lettingsRelief)
+            _.privateResidentsRelief,
```

Comment:
if PRR was greater than zero, then this should still be `lettingsRelief`

Reply:
ooooh yess. good point.

Reply:
Hmmmm... I might change the signature though, as now this time, the Call will be dependant on the required parameter.

Reply:
I think you probably to change it to something like `ReliefDetailsAnswers => Boolean` to avoid wart remover complaining about products and having something like this:
```scala
 hasRequiredPreviousAnswer = { answers =>
   if(privateResidentsReliefIsZero) answers.fold(_.privateResidentsRelief, ...).isDefined  
   else answers.fold(_.lettingsRelief, ...).isDefined
  },
redirectIfNoPreviousAnswers = { answers =>
     if(privateResidentsReliefIsZero) privateResidentsReliefCall  
   else lettingReliefCall
}

```

Reply:
Why not to ReliefDetailsAnswers => Call? Thanks

Reply:
Done, there seems to be many scenarios with the new approach. Maybe we might need to discuss it. But done.